model-optimizer/extensions/middle/reverse_tensor_iterator.py

```diff
@@ -92,23 +131,98 @@ def replace_pattern(self, graph: Graph, match: dict):
             # we can not merge ReverseSequence without equal sequences
             return
 
-        # Modify stride in TI
-        for port_map in [ti.input_port_map, ti.output_port_map]:
-            for port in port_map:
-                if 'axis' in port and port['axis'] is not None and 'external_port_id' in port:
-                    assert port['axis'] == direct_reverse.seq_axis, \
-                        'axis == {} != {} == direct_reverse.seq_dim'.format(port['axis'], direct_reverse.seq_axis)
-                    if 'stride' not in port or port['stride'] is None:
-                        port['stride'] = 1
-                    assert port['stride'] in [-1, 1]
-                    port['stride'] = -port['stride']
-                    if port['stride'] == -1:
-                        port['start'] = -1
-                        port['end'] = 0
-                    elif port['stride'] == 1:
-                        port['start'] = None
-                        port['end'] = None
+        update_ti(ti, direct_reverse)
 
         # Remove reverses
         remove_op_node_with_data_node(graph, direct_reverse)
         remove_op_node_with_data_node(graph, inverse_reverse)
+
+
+class ReverseTensorIteratorLSTMWithSqueeze(MiddleReplacementPattern):
+    """ Fuses Reverse operations around TI: ReverseSequence --> Squeeze --> TI --> UnSqueeze --> ReverseSequence.
+
+        WARNING This transformation is limited to support of very special case of TI but
+        code doesn't check all the cases.
```

Comment:
Can this comment be more detailed? Which case does this transformation support and what are limitations?

Reply:
reformulate comment a little bit